pkg/kafka/encoding.go

```diff
@@ -181,11 +181,16 @@ func (d *Decoder) Decode(data []byte) (logproto.Stream, labels.Labels, error) {
 
 // DecodeWithoutLabels converts a Kafka record's byte data back into a logproto.Stream without parsing labels.
 func (d *Decoder) DecodeWithoutLabels(data []byte) (logproto.Stream, error) {
-	d.stream.Entries = d.stream.Entries[:0]
-	if err := d.stream.Unmarshal(data); err != nil {
+	if len(data) == 0 {
+		return logproto.Stream{}, fmt.Errorf("empty data received")
```

Comment:
There are no args so `errors.New` 👍 

```suggestion
		return logproto.Stream{}, errors.New("empty data received")
```